src/Microsoft.TemplateEngine.Cli/TemplateResolution/TemplateMatchInfoExtensions.cs

```diff
@@ -202,12 +202,52 @@ public static bool IsPartialMatchExceptContext(this ITemplateMatchInfo templateM
 
         public static bool HasNameMatchOrPartialMatch(this ITemplateMatchInfo templateMatchInfo)
         {
-            return templateMatchInfo.MatchDisposition.Any((x => (x.Location == MatchLocation.Name || x.Location == MatchLocation.ShortName) && (x.Kind == MatchKind.Exact || x.Kind == MatchKind.Partial)));
+            return templateMatchInfo.MatchDisposition.Any((x => (x.Location == MatchLocation.Name || x.Location == MatchLocation.ShortName || x.Location == MatchLocation.Classification) && (x.Kind == MatchKind.Exact || x.Kind == MatchKind.Partial)));
```

Comment:
Method name doesn't represent the functionality. Maybe rename the method?

Is there a chance that we broke some other code that was calling this method? Or did we check all the places that this method was called from?

Reply:
This method is only used once. Thinking more on that I'm not sure if partial matches should include match by classification only. I need to check it with PM.

Reply:
Sounds good. In any case, I think someone reading this method name might think it will only return true in case of a name match, but it is also returning true if there is a classification match.